File: c++/remove-element.cpp
//https://oj.leetcode.com/problems/remove-element/

class Solution {
public:
    int removeElement(int A[], int n, int elem) {
        int k = 0;
        for (int i = 0; i < n; i++) {
            if (A[i] != elem) A[k++] = A[i];
        }
        return k;
    }
};

File: python/remove-element.py
class Solution:
    # @param    A       a list of integers
    # @param    elem    an integer, value need to be removed
    # @return an integer
    def removeElement(self, A, elem):
        i = 0
        for j in xrange(len(A)):
            if A[j] != elem:
                A[i] = A[j]
                i += 1

        return i
